.github/workflows/inspektor-gadget.yml

```
@@ -697,6 +697,46 @@ jobs:
           rm -rf /tmp/.buildx-cache
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
 
+  analyze-and-build:
+    name: IG binary size analysis
+    runs-on: ubuntu-latest
+    needs: 
+      - build-ig 
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@3041bf56c941b39c61721a86cd11f3bb1338122a # v5.2.0
+        with:
+          go-version: '1.23.4'
+          cache: true
+        id: go
+      - name: Install go-size-analyzer
+        run: |
+          go install github.com/Zxilly/go-size-analyzer/cmd/gsa@e47aa2561e70b937f2be7441b2eeca984ff60e8b
+      - name: Get ig-linux-amd64.tar.gz from artifact.
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: ig-linux-amd64-tar-gz
+          path: /home/runner/work/inspektor-gadget/
+      - name: Unpack ig-linux-amd64.tar.gz
+        run: |
+          tar zxvf /home/runner/work/inspektor-gadget/ig-linux-amd64.tar.gz
+      - name: Generate JSON
+        run: |
+          gsa ./ig -f "json" --compact > out.json
+      - name: Extract Packages as JSON Array
+        run: |
+          jq '[.packages | to_entries[] | {package: .key, size: .value.size}] | sort_by(.size) | reverse' out.json > packages.json
```

The filename `packages.json` is often used in Javascript projects, so I think it is better to name it differently because we don't use the same syntax as in those JS projects. I suggest the `ig-` prefix, in case we later analyse the binary size of other binaries (e.g. gadgetctl, kubectl-gadget).

Also, I suggest to use `tee` to add the output in the job logs, to be able to debug it later.

```suggestion
          jq '[.packages | to_entries[] | {package: .key, size: .value.size}] | sort_by(.size) | reverse' out.json | tee ig-packages.json
```